src/commands.ts

```diff
@@ -0,0 +1,93 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import { DotNetInfo, requireDotNetSdk, runDotNetCommand } from './dotnet';
+
+export default function registerCommands(dotNetSdkVersion: string): vscode.Disposable[] {
+    return [
+        vscode.commands.registerCommand('build.all.pgproj', async () => {
+            requireDotNetSdk(dotNetSdkVersion).then(buildAllProjects);
+        }),
```

Comment:
Does this happen at startup? I wonder if it would be possible / user friendly to require the .NET SDK when user tries to execute the new command, and not when the command is registered. The extension did not require .net before. Only new commands require it. So, if someone has already been using it and they do not need the new features, this may be frustrating. Just something to consider...